src/awx_plugins/credentials/azure_kv.py

```diff
@@ -72,24 +94,71 @@
     ],
     'required': [
         'url',
-        'client',
-        'secret',
-        'tenant',
         'secret_field',
     ],
 }
 
 
-def azure_keyvault_backend(**kwargs):
-    csc = ClientSecretCredential(
-        tenant_id=kwargs['tenant'],
-        client_id=kwargs['client'],
-        client_secret=kwargs['secret'],
-    )
-    kv = SecretClient(credential=csc, vault_url=kwargs['url'])
-    return kv.get_secret(
-        name=kwargs['secret_field'],
-        version=kwargs.get('secret_version', ''),
+def _initialize_credential(
+    tenant: str = '',
+    client: str = '',
+    secret: str = '',
+) -> TokenCredential:
+    explicit_credentials_provided = all((tenant, client, secret))
+
+    if explicit_credentials_provided:
+        return ClientSecretCredential(
+            tenant_id=tenant,
+            client_id=client,
+            client_secret=secret,
+        )
+
+    try:
+        return ManagedIdentityCredential()
+    except CredentialUnavailableError as cred_err:
+        raise LookupError(
+            'Failed to detect credentials from environment.',
+        ) from cred_err
+
+
+# WPS211 "too many args" is controlled externally
+# pylint: disable-next=too-many-arguments,too-many-positional-arguments
+def azure_keyvault_backend(  # noqa: WPS211
+    *,
+    url: str,
+    client: str,
+    secret: str,
+    tenant: str,
+    secret_field: str,
+    secret_version: str,
+) -> str | None:
+    """Get a credential and retrieve a secret from an Azure Key Vault. An
```

Comment:
The title line is just a single short sentence. The description goes into a separate paragraph per PEP 257.

Reply:
Also, `param` is not a complete word so the spell checker rightfully flags it.